api/features/feature_segments/tests/test_views.py

```diff
@@ -194,6 +175,24 @@ def test_priority_of_multiple_feature_segments(
     assert json_response[0]["id"] == feature_segment.id
     assert json_response[1]["id"] == another_feature_segment.id
 
+    expected_user_id = (
+        None if "HTTP_AUTHORIZATION" in client._credentials else admin_user.id
```

Comment:
I don't love that we have to access `client._credentials` here but (a) I'm not really sure why they're 'private' in a fixture like this and (b) the alternative is splitting this out into separate tests which seems unnecessary.

Reply:
Another option would be to pass a boolean to the function, something like:

```
@pytest.mark.parametrize(
    "client, expect_user_id", [(lazy_fixture("master_api_key_client"), False), (True, lazy_fixture("admin_client"))]
)

```

Reply:
Yeah, that's probably slightly cleaner. Will do that.

Reply:
(assuming I rollback the change I made to move the task call to models)